### PR TITLE
209 new test for reading and writing

### LINKED ISSUE #52: TransformStream ponyfill should only be used when necessary
We may want to change Conflux to not overwrite the `TransformStream` variable if it's defined, e.g.

```js
import ponyfill from 'web-streams-polyfill/ponyfill';
const TransformStream = window.TransformStream || ponyfill.TransformStream
```

### PR DESCRIPTION
**Pull request by jimmywarting**
Written some new test so we don't fu*k up something later when we change stuff. the test are mostly written with newest chrome but we should test other browsers as well.

Also resolves #52 - should do more testing using polyfilled and native streams

thinking we maybe should remove the examples and streamsaver. can do them on jsfiddle or something like it. (means less to maintain)

### CODE REVIEW

**Comment by jimmywarting** [comment]
test file maybe shouldn't live in `src/`

**Comment by bencmbrook** [comment]
I think tests in src are alright. We can fix the coverage report in another PR.

**Comment by jimmywarting** [comment]
how do i fix `The command "yarn test:browserstack" exited with 1.`

**Comment by bencmbrook** [comment]
Looks like Karma test runner is timing out on "All in one big test". I'll extend it 👍

File: test/fixture/build.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const mod = () => ((types, b64) => ({
+  replacer(key) {
+    var val = this[key]
+    return ArrayBuffer.isView(val) ? {$d: b64.encode(new Uint8Array(val))} : val
+  },
+  reviver: (key, val) =>
+      val === null && val !== 'object' ? val:
+      val.$d ? new Blob(b64.decode(val)):
+      val
```
Comment (bencmbrook) [inline test/fixture/build.js:11]: looks like the generated output (`fixture.test.js`) was edited and committed. so when you run based on the pulled file, it works, but if you rebuild fixtures, it breaks. fixing in my PR https://github.com/transcend-io/conflux/pull/54/files

**Comment by bencmbrook** [comment]
There was a deeper bug which has been fixed in #54

**Comment by bencmbrook** [comment]
Closing in favor of #66

**Pull request merged**